### PR TITLE
feat(designer): Split/Refactor FloatingActionMenu, adding FloatingActionMenuOutputs editor

### PR DESCRIPTION
**Pull request by wifun**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

PAuto's response operations, ex. `Respond to a Powerapp or Flow` requires an editor similar to FloatingActionMenu. However it needs to allow tokens, also the added parameter names have slightly different serialization.

This PR consolidates the common code btw. the 2 editors (FloatingActionMenuInputs and FLoatingActionMenuOutputs), primarly UI.
Custom deserialization/serialization/validation is added for the new editor, leveraging `editorViewModel`.

- **What is the new behavior (if this is a feature change)?**

Those that wish to use the new editor will add something along the lines of the following to their manifest:
```
schema: {
  type: 'object',
  'x-ms-editor': 'floatingactionmenu',
  'x-ms-editor-options': {
    supportedTypes: ['text', 'file', 'email', 'boolean', 'number', 'date'],
    menuKind: 'outputs',
  },
  default: {
    type: 'object',
    properties: {},
  },
},
``` 

- **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

No, usage of the new editor is controlled by a manifest `x-ms-editor-options`

- **Please Include Screenshots or Videos of the intended change**:

New editor:
image
image

Editor in v1:
image

AB#24944679

### CODE REVIEW

**Comment by wifun** [comment]
Will add screenshots and do more validation tomorrow 8/25

In the meantime, early feedback is welcome

File: libs/designer-ui/src/lib/dynamicallyaddedparameter/helper.ts [hunk 1]
```diff
@@ -1,11 +1,7 @@
-import type { DynamicallyAddedParameterProps, DynamicallyAddedParameterTypeType, IDynamicallyAddedParameterProperties } from '.';
-import { DynamicallyAddedParameterType } from '.';
-import constants from '../constants';
-import type { ValueSegment } from '../editor';
-import { ValueSegmentType } from '../editor';
+import type { DynamicallyAddedParameterTypeType } from '../dynamicallyaddedparameter';
```
Comment (wifun) [inline libs/designer-ui/src/lib/dynamicallyaddedparameter/helper.ts:1]: Most of the code here moved to `libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenuinputs/helper.ts`.

Only notable small change is to `generateDynamicParameterKey()`

File: libs/designer-ui/src/lib/dynamicallyaddedparameter/index.tsx [hunk 2]
```diff
@@ -13,48 +13,24 @@ export const DynamicallyAddedParameterType = {
...
   schemaKey: string;
-  properties: IDynamicallyAddedParameterProperties;
-  required: boolean;
-  onChange?: TextFieldOnChangeHandler;
-  onDelete?: DeleteDynamicallyAddedParameterHandler;
+  icon: string;
+  title: string;
+  onTitleChange: (schemaKey: string, newValue: string | undefined) => void;
+  onDelete: (schemaKey: string) => void;
+  onRenderValueField: (schemaKey: string) => JSX.Element;
```
Comment (wifun) [inline libs/designer-ui/src/lib/dynamicallyaddedparameter/index.tsx:22]: Overall changes to this file is to make this component more customizable

File: libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+            return renderMenuItem(item);
+          })}
+        </div>
+      </div>
+    );
+  };
+
+  return (
+    <>
+      <div className="msla-dynamic-added-params-container">{props.children}</div>
```
Comment (wifun) [inline libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx:143]: This file is pretty much all the UI code (the main similarity btw. FloatingActionMenuInputs and Outputs)

File: libs/designer/src/lib/core/utils/parameters/helper.ts [hunk 4]
```diff
@@ -152,6 +153,12 @@ import type { Dispatch } from '@reduxjs/toolkit';
 
 // import { debounce } from 'lodash';
 
+// TODO(WIFUN): Why does export from designer-ui no work?
```
Comment (wifun) [inline libs/designer/src/lib/core/utils/parameters/helper.ts:156]: I intend to remove all TODOs before merging PR

File: Localize/lang/strings.json [hunk 5]
```diff
@@ -126,6 +126,7 @@
   "3X4FHS": "Choose the type of user input",
   "3Y8a6G": "Required parameters {parameters} not set or invalid",
   "3cZZKj": "Divider",
+  "3e19i+": "Enter a value to respond",
```
Comment (ek68794998) [inline Localize/lang/strings.json:129]: "Enter a value to respond with"?

File: libs/designer/src/lib/core/utils/parameters/helper.ts [hunk 6]
```diff
@@ -2174,11 +2199,53 @@ function getStringifiedValueFromEditorViewModel(parameter: ParameterInfo, isDefi
...
+  editorViewModel: FloatingActionMenuOutputViewModel
+): string | undefined => {
+  const value: typeof editorViewModel.schema & { additionalProperties?: { outputValueMap?: Record<string, unknown> } } = clone(
+    editorViewModel.schema
+  );
+  const schemaProperties: typeof editorViewModel.schema.properties = {};
+  const outputValueMap: Record<string, unknown> = {};
+
+  // We want to cast (for example) "1" to 1 if the dynamically added parameter type is 'Number'
+  const commonProperties = { supressCasting: false, info: parameter.info };
```
Comment (wifun) [inline libs/designer/src/lib/core/utils/parameters/helper.ts:2225]: Doesn't seem like this does what I want ("1" to 1), searching for a helper that will do this cast....

File: libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+      e.preventDefault();
+      e.stopPropagation();
+      toggleExpanded();
+    }
+  };
+
+  const renderMenuButton = (): JSX.Element => {
+    return (
+      <div role="button" className="msla-floating-action-menu" onClick={toggleExpanded} onKeyDown={toggleExpandedOnKeyDown} tabIndex={0}>
+        <span className="msla-floating-action-menu-plus-icon">{'+'}</span>
```
Comment (ek68794998) [inline libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx:46]: Do we have actual Fluent icons we can use here?

File: libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+        <span className="msla-floating-action-menu-items-title">{props.expandedTitle}</span>
+        <span
+          role="button"
+          aria-label={closeErrorButtonAriaLabel}
+          className="msla-floating-action-menu-items-close"
+          onClick={toggleExpanded}
+          onKeyDown={toggleExpandedOnKeyDown}
+          tabIndex={0}
+        >
+          {'x'}
```
Comment (ek68794998) [inline libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx:69]: Fluent icon?

**Review by rllyy97 — APPROVED** [review]

File: libs/designer-ui/src/lib/dynamicallyaddedparameter/helper.ts [hunk 9]
```diff
@@ -111,292 +107,9 @@ function getDescriptionForDynamicallyAddedParameterType(type: DynamicallyAddedPa
...
-    'x-ms-content-hint': itemType,
-    'x-ms-dynamically-added': true,
-  };
-}
-
-export function generateDynamicParameterKey(
-  parameters: DynamicallyAddedParameterProps[],
-  typeHint: DynamicallyAddedParameterTypeType
-): string {
-  const currentKeys = parameters.map((parameter) => parameter.schemaKey);
```
Comment (rllyy97) [inline libs/designer-ui/src/lib/dynamicallyaddedparameter/helper.ts:315]: What's the purpose of removing this map?  Can we make it 
`const currentKeys = parameters.map((parameter) => parameter?.schemaKey ?? parameter);`

File: libs/designer/src/lib/core/utils/parameters/helper.ts (hunk 6 above)
Comment (wifun) [reply libs/designer/src/lib/core/utils/parameters/helper.ts]: using `parameterValueWithoutCasting()`

File: libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx (hunk 7 above)
Comment (wifun) [reply libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx]: swapped to icons

**Pull request merged**